application/controllers/HistoryController.php

```diff
@@ -78,9 +78,11 @@ public function indexAction()
 
         $history->filter(Filter::lessThanOrEqual('event_time', $before));
         $this->filter($history, $filter);
+
+        $history->getWith()['history.host']->setJoinType('LEFT');
         $history->getSelectBase()
-            // Make sure we'll fetch service history entries only for services which still exist
-            ->where(['history.service_id IS NULL', 'history_service.id IS NOT NULL'], Sql::ANY);
+            // Because of LEFT JOINs, make sure we'll fetch history entries only for items which still exist:
+            ->where(['history_host.id IS NOT NULL', 'history_service.id IS NOT NULL'], Sql::ANY);
```

Comment:
What if there's a host (`history_host.id IS NOT NULL`) for the service history entry, but no service? (`history_service.id IS NULL`)